pgeocode.py

```diff
@@ -311,9 +311,61 @@ def query_postal_code(self, codes):
             response = response.iloc[0]
         return response
 
-    def query_location(self, name):
-        """Get locations information from a community/minicipality name"""
-        pass
+    def query_location(self, name: str, top_k: int = 500, fuzzy_threshold: int = 80,
+                       col: str = "place_name") -> pd.DataFrame:
+        """Get location information from a place name
+
+        Parameters
+        ----------
+        name: str
+          string containing place names to search for
+        top_k: int
+          maximum number of results (rows in DataFrame) to return
+        fuzzy_threshold: int
```

Comment:
To make behavior not depend on installed libraries, let's make this None by default. And if it's None, no fuzzy search is done, and the `thefuzz` is not used. 

You could add a sentence above parameters in the description saying that to enable fuzzy search one need to provide the fuzzy_threshold parameter. And then in the description of this parameter mention what are good default value and link to the documentation of thefuzz because right it's not very clear how to interpret this value.

If it's not None

Reply:
Adapted it, let me know if this works for you